modules/rules_foreign_cc/0.7.1/patches/label.patch

```diff
@@ -0,0 +1,159 @@
+diff --git a/foreign_cc/built_tools/make_build.bzl b/foreign_cc/built_tools/make_build.bzl
+index 1d0ad29..cc15a55 100644
+--- a/foreign_cc/built_tools/make_build.bzl
++++ b/foreign_cc/built_tools/make_build.bzl
+@@ -103,7 +103,7 @@ make_tool = rule(
+     output_to_genfiles = True,
+     implementation = _make_tool_impl,
+     toolchains = [
+-        str(Label("//foreign_cc/private/framework:shell_toolchain")),
++        "@rules_foreign_cc//foreign_cc/private/framework:shell_toolchain",
```

Comment:
I think you should be able to lose the `@rules_foreign_cc` here, too, since these labels are resolved within the context of the current .bzl file (unlike labels within macros, which are resolved within the context of the calling package). Same below mostly, I've marked exceptions out.

Reply:
It should indeed work, just keep in mind that it doesn't work with Bazel 4 due to the lack of https://github.com/bazelbuild/bazel/commit/d23e0a1bc75510ac92ecdc4628ada33cdaa1d53a. So while it's a good change to make in a BCR patch, it can't be merged back upstream without breaking compatibility with Bazel 4.

Reply:
That's true. You could instead always keep the `Label(...)` call, which will forcably resolve the label within the current .bzl file, no matter where this is (in a macro, in a rule, outside either, etc). (N.B. this comes with its own caveats due to https://github.com/bazelbuild/bazel/issues/12963; in this case we're good since we can just do without a `@repo` part.)

tl;dr: use `Label("//foreign_cc/private/framework:shell_toolchain")`. It should work on all versions.

Reply:
Unfortunately using  `Label("//foreign_cc/private/framework:shell_toolchain")` doesn't work on 4.2.2:
```
ERROR: Traceback (most recent call last):
	File "/home/ubuntu/rules_foreign_cc/foreign_cc/built_tools/ninja_build.bzl", line 28, column 18, in <toplevel>
		ninja_tool = rule(
Error in rule: at index 0 of toolchains, got element of type Label, want string
```

Reply:
ah, dang it. The `toolchains` parameter of `rule` doesn't work like the `default` parameter of `attr.label`. A shame. Please ignore me then!